File: src/001.Array/035.SearchInsertPosition/35.搜索插入位置.go
/*
 * @lc app=leetcode.cn id=35 lang=golang
 *
 * [35] 搜索插入位置
 */

// @lc code=start
func searchInsert(nums []int, target int) int {
	l := 0
	r := len(nums)

	for l < r {
		mid := l + (r - l) / 2

		if nums[mid] == target {
			return mid
		}

		if nums[mid] < target {
			l = mid + 1
		} else {
			r = mid
		}
	}

	return l
}
// @lc code=end

File: src/001.Array/035.SearchInsertPosition/35.搜索插入位置.js
/*
 * @lc app=leetcode.cn id=35 lang=javascript
 *
 * [35] 搜索插入位置
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
  let index = 0;

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] == target) return i;

    if (nums[i] < target) {
      index = i + 1;
    }
  }

  return index;
};
// @lc code=end

